src/Modules/Data/Commands.Data.Test/GetPowerBIDatasourceTests.cs

```diff
@@ -0,0 +1,160 @@
+            {
+                // Arrange
+                ProfileTestUtilities.SafeDisconnectFromPowerBI(ps);
+                var parameters = new Dictionary<string, object>()
+                {
+                    { nameof(GetPowerBIDatasource.DatasetId), Guid.NewGuid() },
+                    { nameof(GetPowerBIDatasource.WorkspaceId), Guid.NewGuid() },
+                    { nameof(GetPowerBIDatasource.Scope), PowerBIUserScope.Organization }
+                };
+                ps.AddCommand(GetPowerBIDatasourceCmdletInfo).AddParameters(parameters);
+
+                // Act
+                var results = ps.Invoke();
+
+                // Assert
+                Assert.Fail("Should not have reached this point");
+            }
+        }
+
+        [TestMethod]
+        public void GetPowerBIDatasourceWithIndividual()
+        {
+            // Arrange
+            var datasetId = Guid.NewGuid();
+            var expectedDatasources = new List<Datasource> { new Datasource { DatasourceId = Guid.NewGuid().ToString(), Name = "TestDatasource", GatewayId = Guid.NewGuid().ToString() } };
+            var client = new Mock<IPowerBIApiClient>();
+            client.Setup(x => x.Datasets.GetDatasources(datasetId, null)).Returns(expectedDatasources);
+            var initFactory = new TestPowerBICmdletInitFactory(client.Object);
+            var cmdlet = new GetPowerBIDatasource(initFactory)
+            {
+                Scope = PowerBIUserScope.Individual,
+                DatasetId = datasetId,
+                ParameterSet = "List",
+            };
+
+            // Act
+            cmdlet.InvokePowerBICmdlet();
+
+            // Assert
+            initFactory.AssertExpectedUnitTestResults(expectedDatasources);
+        }
+
+        [TestMethod]
+        public void GetPowerBIDatasourceWithIndividualAndWorkspaceId()
+        {
+            // Arrange
+            var datasetId = Guid.NewGuid();
+            var workspaceId = Guid.NewGuid();
+            var expectedDatasources = new List<Datasource> { new Datasource { DatasourceId = Guid.NewGuid().ToString(), Name = "TestDatasource", GatewayId = Guid.NewGuid().ToString() } };
+            var client = new Mock<IPowerBIApiClient>();
+            client.Setup(x => x.Datasets.GetDatasources(datasetId, workspaceId)).Returns(expectedDatasources);
+            var initFactory = new TestPowerBICmdletInitFactory(client.Object);
+            var cmdlet = new GetPowerBIDatasource(initFactory)
+            {
+                Scope = PowerBIUserScope.Individual,
+                DatasetId = datasetId,
+                WorkspaceId = workspaceId,
+                ParameterSet = "List",
+            };
+
+            // Act
+            cmdlet.InvokePowerBICmdlet();
+
+            // Assert
+            initFactory.AssertExpectedUnitTestResults(expectedDatasources);
+        }
+
+        [TestMethod]
+        public void GetPowerBIDatasourceWithOrganization()
```

Comment:
Going based off AddPowerBIWorkspaceUserTests as an example, I would have named this test, "GetPowerBIDatasourceOrganizationScope_ListParameterSet".